main.js

```diff
@@ -1 +1,15 @@
-// all your js goes here
+$(document).ready(function(){
+	$('form#add').on('change', '#image-upload', function() {
```

Comment:
it's good that you understand event delegation, but it's not necessary here. the benefit of delegation comes when there is more than one element that may emit an event. here there is only one, so it is probably more performant to just listen to that element.